libs/ledger-live-common/src/device/use-cases/appDataBackup/deleteAppData.ts

```diff
@@ -0,0 +1,54 @@
+import { catchError, from, Observable, of, switchMap, throwError } from "rxjs";
+import { DeviceModelId } from "@ledgerhq/devices";
+import {
+  AppName,
+  AppStorageType,
+  DeleteAppDataError,
+  DeleteAppDataEvent,
+  DeleteAppDataEventType,
+  StorageProvider,
+} from "./types";
+
+export function deleteAppData(
+  appName: AppName,
+  deviceModelId: DeviceModelId,
+  storageProvider: StorageProvider<AppStorageType>,
+): Observable<DeleteAppDataEvent> {
+  const obs = new Observable<DeleteAppDataEvent>(subscriber => {
+    subscriber.next({ type: DeleteAppDataEventType.AppDataDeleteStarted });
+    const sub = from(storageProvider.getItem(`${deviceModelId}-${appName}`))
```

Comment:
[SHOULD] Have 1 function (unique source of truth) `getAppDataStorageId` (or similar naming) to get the string `${deviceModelId}-${appName}`, reused throughout backup/delete/restore.